app/forms/common/Date.scala

```diff
@@ -51,39 +52,41 @@ object Date {
   val monthKey = "month"
   val dayKey = "day"
 
-  private val correctDay: Int => Boolean = (day: Int) => day >= 1 && day <= 31
-  private val correctMonth: Int => Boolean = (month: Int) => month >= 1 && month <= 12
-  private val correctYear: Int => Boolean = (year: Int) => year >= 2000 && year <= 2099
-
   val mapping: Mapping[Date] = {
 
-    def validate(day: Option[Int], month: Option[Int], year: Option[Int]): Boolean =
-      (day, month, year) match {
-        case (Some(d), Some(m), Some(y)) => Try(LocalDate.of(y, m, d)).isSuccess
-        case _                           => false
-      }
+    def build(day: Try[Int], month: Try[Int], year: Try[Int]): Try[LocalDate] =
+      for {
+        d <- day
+        m <- month
+        y <- year
+        date <- Try(LocalDate.of(y, m, d))
+      } yield date
```

Comment:
Can we move Try to yield?
```
for {
  d <- day
  m <- month
  y <- year
} yield Try(LocalDate.of(y, m, d)
```

Reply:
No i could not this would lead to `Try[Try[Int]]` optionally i can remove try and LocalDate but i'am not sure how it would handle exceptions